embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/EmbraceApiService.kt

```diff
@@ -53,33 +52,39 @@ internal class EmbraceApiService(
      * @return a future containing the configuration.
      */
     @Throws(IllegalStateException::class)
+    @Suppress("UseCheckOrError")
     override fun getConfig(): RemoteConfig? {
         var request = prepareConfigRequest(configUrl)
         val cachedResponse = cachedConfigProvider(configUrl, request)
         if (cachedResponse.isValid()) { // only bother if we have a useful response.
             request = request.copy(eTag = cachedResponse.eTag)
         }
-        val response = apiClient.executeGet(request)
 
-        return when (response.statusCode) {
-            HttpURLConnection.HTTP_OK -> {
+        return when (val response = apiClient.executeGet(request)) {
```

Comment:
Does it make sense just to preserve the existing behaviour when dealing with how to handle a reponses and change this for real when you add the code the handle 429?